server/src/test/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolverTests.java

```diff
@@ -1810,6 +1828,125 @@ public void testIgnoreThrottled() {
         }
     }
 
+    public void testSystemIndexResolutionWhenAllowed() {
+        Settings settings = Settings.builder().build();
+        Metadata.Builder mdBuilder = Metadata.builder()
+            .put(indexBuilder(".ml-meta", settings).state(State.OPEN).system(true))
+            .put(indexBuilder(".watches", settings).state(State.OPEN).system(true))
+            .put(indexBuilder(".ml-stuff", settings).state(State.OPEN).system(true))
+            .put(indexBuilder("some-other-index").state(State.OPEN));
+        ClusterState state = ClusterState.builder(new ClusterName("_name")).metadata(mdBuilder).build();
+
+        // Single name
+        {
+            SearchRequest request = new SearchRequest(".ml-meta");
+
+            List<String> indexNames = Arrays
+                .stream(indexNameExpressionResolver.concreteIndices(state, request))
+                .map(i -> i.getName())
+                .collect(Collectors.toList());
```

Comment:
What do you think about defining a helper method, to cut down on the ceremony here? Something like:

```java
private void getConcreteIndexNames(ClusterState state, SearchRequest request) {
    return List<String> indexNames = Arrays
        .stream(indexNameExpressionResolver.concreteIndices(state, request))
        .map(i -> i.getName())
        .collect(Collectors.toList());
}
```